src/main/java/KeyCommand.java

```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+
+public class KeyCommand extends Command {
+    protected String keyword;
+
+    public KeyCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.showLine("Here are the matching tasks in your list: ");
+        ArrayList<String> resultList = new ArrayList<String>();
+
+        for (int i = 0; i < tasks.size(); i++) {
+            String[] taskDesc = tasks.get(i).getDesc().split(" ");
+
+            for (int j = 0; j < taskDesc.length; j++) {
```

Comment:
I agree with Jonathan, contains capture the logic more cleanly as compared to equals. Maybe you can consider modifying!